api/src/main/java/ca/bc/gov/educ/api/gradalgorithm/service/GradAlgorithmService.java

```diff
@@ -216,21 +216,12 @@ private void getListOfOptionalProgramStatus(String optionalProgramCode,OptionalP
 	}
 
 	private  void processGraduation(GradAlgorithmOptionalStudentProgram gradStudentOptionalAlg, RuleProcessorData ruleProcessorData) {
+		Date gradDate = new Date();
+
 		if (gradStudentOptionalAlg.isOptionalGraduated() && ruleProcessorData.isGraduated()) {
-			String mainProgramCompletionDate;
-			if (ruleProcessorData.getGradStatus().getProgramCompletionDate().length() > 7) {
-				mainProgramCompletionDate = ruleProcessorData.getGradStatus().getProgramCompletionDate();
-			}else {
-				mainProgramCompletionDate = APIUtils.parsingTraxDate(ruleProcessorData.getGradStatus().getProgramCompletionDate());
-			}
-			if (!gradStudentOptionalAlg.getOptionalStudentCourses().getStudentCourseList().isEmpty()) {
-				String optionalPrgComlDate = getGradDate(gradStudentOptionalAlg.getOptionalStudentCourses().getStudentCourseList(),
-						gradStudentOptionalAlg.getOptionalStudentAssessments().getStudentAssessmentList());
-				gradStudentOptionalAlg.setOptionalProgramCompletionDate(optionalPrgComlDate == null ? mainProgramCompletionDate:optionalPrgComlDate);
-			} else {
-				gradStudentOptionalAlg.setOptionalProgramCompletionDate(
-						getGradDate(ruleProcessorData.getStudentCourses(), ruleProcessorData.getStudentAssessments()));
-			}
+
+			String result = new SimpleDateFormat(DEFAULT_DATE_FORMAT).format(gradDate);
```

Comment:
Hi Mamatha, you can just instantiate the Date object in the format method for less code. Just change to: String result = new SimpleDateFormat(DEFAULT_DATE_FORMAT).format(new Date());